test/test_fourier.jl

```diff
@@ -0,0 +1,419 @@
+    ft1 = fft(data1)
+    ft2 = fft(data2)
+    dt = 0.01
+    N = length(data)
+    mean = Statistics.mean(data)
+    meanrate = mean / dt
+    freq = fftfreq(length(data), 1/dt)
+    good = (freq .> 0 .&& freq .< 0.1)
+    ft1, ft2 = ft1[good], ft2[good]
+    cross = normalize_periodograms(
+        ft1 .* conj(ft2), dt, N, mean_flux = mean, norm="abs", power_type="all")
+    pds1 = normalize_periodograms(
+        ft1 .* conj(ft1), dt, N, mean_flux = mean, norm="abs", power_type="real")
+    pds2 = normalize_periodograms(
+        ft2 .* conj(ft2), dt, N, mean_flux = mean, norm="abs", power_type="real")
+
+    p1noise = poisson_level("abs",meanrate=meanrate)
+    p2noise = poisson_level("abs",meanrate=meanrate)
+
+    @testset "test_intrinsic_coherence" begin
+        coh = estimate_intrinsic_coherence(
+            cross, pds1, pds2, p1noise, p2noise, N)
+        @test all(.≈(coh, 1, atol=0.001))
+    end
+
+    @testset "test_raw_high_coherence" begin
+        coh = raw_coherence(cross, pds1, pds2, p1noise, p2noise, N)
+        @test all(.≈(coh, 1, atol=0.001))
+    end
+
+    @testset "test_raw_low_coherence" begin
+        nbins = 2
+        C, P1, P2 = cross[1:nbins], pds1[1:nbins], pds2[1:nbins]
+        bsq = bias_term(P1, P2, p1noise, p2noise, N)
+        # must be lower than bsq!
+        low_coh_cross = rand.(Normal.(bsq.^0.5 / 10, bsq.^0.5 / 100)) .+ 0.0im
+        coh = raw_coherence(low_coh_cross, P1, P2, p1noise, p2noise, N)
+        @test all(coh .== 0)
+    end
+
+    @testset "test_raw_high_bias" begin
+        C = [12986.0 + 8694.0im]
+        P1 = [476156.0]
+        P2 = [482751.0]
+        P1noise = 495955
+        P2noise = 494967
+    
+        coh = raw_coherence(C, P1, P2, P1noise, P2noise, 499; intrinsic_coherence=1)
+        coh_sngl = raw_coherence(C[1], P1[1], P2[1], P1noise, P2noise, 499; intrinsic_coherence=1)
+        @test coh==real(C .* conj(C))./ (P1 .* P2)
+        @test coh_sngl==real(C .* conj(C))[1] / (P1[1] * P2[1])
+    end
+    
+end
+
+@testset "test_fourier" begin
+    dt = 1
+    len = 100
+    ctrate = 10000
+    N = Int64(len / dt)
+    dt = len / N
+    times = sort(rand(Uniform(0, len), len * ctrate))
+    gti = [[0 len];;]
+    bins = LinRange(0, len, N + 1)
+    counts = fit(Histogram, times, bins).weights
+    errs = fill!(similar(counts),1) * sqrt(ctrate)
+    bin_times = (bins[1:end-1] + bins[2:end]) / 2
+    segment_size = 20.0
+    times2 = sort(rand(Uniform(0, len), len * ctrate))
+    counts2 = fit(Histogram, times2, bins).weights
+    errs2 = fill!(similar(counts2),1) * sqrt(ctrate)
+
+    @test get_average_ctrate(times, gti, segment_size) == ctrate
+    @test get_average_ctrate(bin_times, gti, segment_size; counts=counts) == ctrate
+
+    @testset "test_fts_from_segments_cts_and_events_are_equal" begin
+        N = Int64(round(segment_size / dt))
+        fts_evts = [
+            f for f in get_flux_iterable_from_segments(times, gti, segment_size, n_bin=N)
+        ]
```

Comment:
Wouldn't this be equivalent to
```suggestion
        fts_evts = collect(get_flux_iterable_from_segments(times, gti, segment_size; n_bin=N))
```
?  Same below